### ISSUE TITLE
Chapter 4 Exercise 12 - Math Latex display

### ISSUE THREAD

**Issue by vidyabhandary** [ageron/handson-ml2#93]
In Chapter 4 - Exercise: Batch Gradient Descent with early stopping for Softmax Regression, the math equations are not getting displayed correctly. I checked in Chrome, Edge and Firefox. 

$J(\mathbf{\Theta}) =

    \dfrac{1}{m}\sum\limits{i=1}^{m}\sum\limits{k=1}^{K}{y_k^{(i)}\log\left(\hat{p}_k^{(i)}\right)}$

And the equation for the gradients:

$\nabla_{\mathbf{\theta}^{(k)}} \, J(\mathbf{\Theta}) = \dfrac{1}{m} \sum\limits_{i=1}^{m}{ \left ( \hat{p}^{(i)}_k - y_k^{(i)} \right ) \mathbf{x}^{(i)}}$

-----

_I think it can be corrected with the following change_ 

-----

So the equations we will need are the cost function:

$$J(\mathbf{\Theta}) =
\dfrac{1}{m}\sum\limits_{i=1}^{m}\sum\limits_{k=1}^{K}{y_k^{(i)}\log\left(\hat{p}_k^{(i)}\right)}$$

And the equation for the gradients:

$$\nabla_{\mathbf{\theta}^{(k)}} \, J(\mathbf{\Theta}) = \dfrac{1}{m} \sum\limits_{i=1}^{m}{ \left ( \hat{p}^{(i)}_k - y_k^{(i)} \right ) \mathbf{x}^{(i)}}$$

-----

Thank you for your awesome book.

**Comment by vidyabhandary** [comment]
Realized the display seems to be only when browsing the GitHub page - it gets displayed via the jupyter environment correctly.

**Comment by ageron** [comment]
Hi @vidyabhandary ,

Thanks for your feedback, and thanks as well for your very kind words, I'm glad you enjoy my book! :)

Indeed, github's notebook viewer is not great when it comes to displaying equations. I don't know what they're doing under the hood, but [it's not working](https://github.com/jupyter/nbviewer/issues/452). That's why I have this warning on the repo's home page:

![image](https://user-images.githubusercontent.com/76661/78009426-7f3d3980-739d-11ea-8ab2-a9dffe25d2eb.png)

Pretty much any other viewer than github's should display things correctly, as you've noticed. You can use Colab, or Jupyter.org's viewer, etc.

Hope this helps.
